src/test/java/com/epam/task/exceptions/PurchaseFactoryTest.java

```diff
@@ -0,0 +1,68 @@
+package com.epam.task.exceptions;
+
+import com.epam.task.exceptions.customExc.CsvLineException;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvFileSource;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class PurchaseFactoryTest {
```

Comment:
it seems to me you made few tests. 
but that's not a big deal for the coding exercise unlike a prod project.